modules/cassandra/src/test/java/org/testcontainers/containers/CassandraContainerTest.java

```diff
@@ -106,8 +106,63 @@ public void testCassandraGetCluster() {
         }
     }
 
+    @Test
+    public void testAuthenticationSupport() {
+        try (
+            CassandraContainer cassandraContainer = new CassandraContainer<>()
+                .withAuthentication()
+                .withConfigurationOverride("cassandra-authentication-test-configuration")
```

Comment:
is configuration override a must? I would expect `withAuthentication()` to do the job, otherwise, what's the point of it?

Reply:
```withAuthentication()``` and ```withConfigurationOverride()``` will go hand in hand. 
```withConfigurationOverride()``` will be used to change the server configuration by changing the line ```authenticator: AllowAllAuthenticator``` to ```authenticator: PasswordAuthenticator```.  As the Cassandra Client will be unaware of this the end user is expected to also set the fact authentication is being used so that the internal Cassandra Client is aware of that. 
The other way to do would be that inside the ```withConfigurationOverride()``` we actually parse the configuration.yml file and check the value of the ```authenticator: ``` variable and then set the authentication boolean within the class using that.
Maybe the second way is better, but it will lead to a bit of messy code. Please advice and I will change accordingly.

Reply:
@bsideup - Please Suggest if you like the second way. I will make and push the changes.

Reply:
Need to think about it. I definitely see an issue with calling `withAuthentication()` when in fact no authentication will be configured unless the user overrides the configuration. And the other way around - when they do, but forget to call `withAuthentication()`...

Also, @kiview @rnorth WDYT?

Reply:
@bsideup - There doesn't seem to any movment on this, thus I have changed the implementation a little. If you have time please take a look and see if you like it.

Reply:
@bsideup, @kiview @rnorth - Would really appreciate a re-review of this. I think authentication in Cassandra is important.

Reply:
Sorry - it's been a long time.

I'm pretty uneasy with this, as @bsideup is. How essential is this?